sentry_sdk/scope.py

```diff
@@ -147,21 +187,145 @@ class Scope(object):
+
         self.clear()
 
         incoming_trace_information = self._load_trace_data_from_env()
         self.generate_propagation_context(incoming_data=incoming_trace_information)
 
+    @classmethod
+    def get_current_scope(cls):
+        # type: () -> Scope
+        """
+        Returns the current scope.
+
+        .. versionadded:: 1.XX.0
+        """
+        current_scope = _current_scope.get()
+        if current_scope is None:
+            current_scope = Scope(ty="current")
+            _current_scope.set(current_scope)
+
+        return current_scope
+
+    @classmethod
+    def get_isolation_scope(cls):
+        # type: () -> Scope
+        """
+        Returns the isolation scope.
+
+        .. versionadded:: 1.XX.0
+        """
+        isolation_scope = _isolation_scope.get()
+        if isolation_scope is None:
+            isolation_scope = Scope(ty="isolation")
+            _isolation_scope.set(isolation_scope)
+
+        return isolation_scope
+
+    @classmethod
+    def get_global_scope(cls):
+        # type: () -> Scope
+        """
+        Returns the global scope.
+
+        .. versionadded:: 1.XX.0
+        """
+        global _global_scope
+        if _global_scope is None:
+            _global_scope = Scope(ty="global")
+
+        return _global_scope
+
+    @classmethod
+    def get_client(cls):
+        # type: () -> Union[sentry_sdk.Client, sentry_sdk.client.NoopClient]
+        """
+        Returns the currently used :py:class:`sentry_sdk.Client`.
+        This checks the current scope, the isolation scope and the global scope for a client.
+        If no client is available a :py:class:`sentry_sdk.client.NoopClient` is returned.
+
+        .. versionadded:: 1.XX.0
+        """
+        client = Scope.get_current_scope().client
+        if client is not None:
+            return client
+
+        client = Scope.get_isolation_scope().client
+        if client is not None:
+            return client
+
+        client = Scope.get_global_scope().client
+        if client is not None:
+            return client
+
+        return NoopClient()
+
+    def set_client(self, client=None):
+        # type: (Optional[sentry_sdk.Client]) -> None
+        """
+        Sets the client for this scope.
+        :param client: The client to use in this scope.
+            If `None` the client of the scope will be replaced by a :py:class:`sentry_sdk.NoopClient`.
+
+        .. versionadded:: 1.XX.0
+        """
+        self.client = client or NoopClient()
+
+    @property
+    def is_forked(self):
+        # type: () -> bool
+        """
+        Weither this scope is a fork of another scope.
+
+        .. versionadded:: 1.XX.0
+        """
+        return self.original_scope is not None
+
+    def fork(self):
+        # type: () -> Scope
+        """
+        Returns a fork of this scope.
+
+        .. versionadded:: 1.XX.0
+        """
+        self.original_scope = self
+        return copy(self)
```

Comment:
This method confuses me a bit. I would expect only the newly created Scope to have `is_forked == True`, but if I my reasoning is correct, both the original scope and the copy will have `is_forked == True` after this method is called.

Is this behavior intentional? If so, why does it make sense for both scopes to be forked?